### PR TITLE
Create request

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2460

### CODE REVIEW

File: Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/MockJSONHttpClient.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            let txData = jsonData("MultiSendApproveMultihopBatchSwapExactIn")
+            do {
+                let output: T.ResponseType = try self.response(from: txData)
+                completion(.success(output))
+            } catch {
+                completion(.failure(error))
+            }
+        } else {
+            print("---> TransactionPreviewRequest | Unexpected request... -> error")
+            //completion(.failure(InvalidRequestError))
```
Comment (biafra23) [inline Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/MockJSONHttpClient.swift]: What is the easiest way to creat a new error class and return it here. For example `InvalidRequestError`?

File: Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        // Will always return the mocked response in completion
+        asyncExecuteMock(request: TransactionPreviewRequest(safe: transaction.safe!,
+                                                               sender: sender,
+                                                               transaction: transaction,
+                                                               chainId: chainId),
+                            completion: completion)
+    }
+
+    @discardableResult
+    func previewTransaction(transaction: Transaction, sender: AddressString, chainId: String) throws -> SCGModels.TransactionDetails? {
```
Comment (biafra23) [inline Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift:71]: This is not mocked. Do we need it?

File: Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift [hunk 3]
```diff
@@ -215,6 +215,23 @@ class WCIncomingTransactionRequestViewController: UIViewController {
 
         tableView.estimatedRowHeight = BasicCell.rowHeight
 
+        // TODO Is this the right place for this call?
+         App.shared.clientGatewayService.asyncPreviewTransaction(
```
Comment (biafra23) [inline Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift]: Is this the right place for this call?

File: Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/MockJSONHttpClient.swift (hunk 1 above)
Comment (DmitryBespalov) [reply Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/MockJSONHttpClient.swift]: for debug issues, just use a String, I think it is defined as an error protocol implementation somewhere, so you can use it

File: Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift (hunk 3 above)
Comment (DmitryBespalov) [reply Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift]: yes I think so

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**